prowler/providers/aws/services/iam/iam_service.py

```diff
@@ -384,9 +384,10 @@ def __list_mfa_devices__(self):
                 for page in list_mfa_devices_paginator.paginate(UserName=user.name):
                     for mfa_device in page["MFADevices"]:
                         mfa_serial_number = mfa_device["SerialNumber"]
-                        mfa_type = (
-                            mfa_device["SerialNumber"].split(":")[5].split("/")[0]
-                        )
+                        try:
+                            mfa_type = mfa_serial_number.split(":")[5].split("/")[0]
+                        except Exception:
```

Comment:
```suggestion
                        except IndexError:
```

Because we want to catch the IndexError if the operation performed with the `mfa_type` fails. We need to start catching specific exceptions instead of a generic one.

Reply:
True! Done ✅